src/node.ts

```diff
@@ -5,32 +5,180 @@ import { analyticsNode } from './plugins/analytics-node'
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+  /**
+   * Records page views on your website, along with optional extra information
+   * about the page viewed by the user.
+   * @param name - The name of the page.
+   * @param properties - A dictionary of properties of the page.
+   * @param options
+   * @param callback
+   */
+  page(
+    name: string,
+    properties: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+  /**
+   * Records page views on your website, along with optional extra information
+   * about the page viewed by the user.
+   * @param category - The category of the page.
+   * Useful for cases like ecommerce where many pages might live under a single category.
+   * @param name - The name of the page.
+   * @param properties - A dictionary of properties of the page.
+   * @param options
+   * @param callback
+   */
+  page(
+    category: string,
+    name: string,
+    properties: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Records screen views on your app, along with optional extra information
+   * about the screen viewed by the user.
+   * @param properties
+   * @param options
+   * @param callback
+   */
+  screen(
+    properties: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+  /**
+   * Records screen views on your app, along with optional extra information
+   * about the screen viewed by the user.
+   * @param name - The name of the screen.
+   * @param properties
+   * @param options
+   * @param callback
+   */
+  screen(
+    name: string,
+    properties: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Records actions your users perform.
+   * @param event - The name of the event you're tracking.
+   * @param properties - A dictionary of properties for the event.
+   * @param options
+   * @param callback
+   */
+  track(
+    event: string,
+    properties: object,
+    options: NodeOptions,
+    callback?: Callback
+  ): Promise<DispatchedEvent>
+
+  /**
+   * Registers one or more plugins to augment Analytics functionality.
+   * @param plugins
+   */
+  register(...plugins: Plugin[]): Promise<Context>
+
+  /**
+   * Deregisters one or more plugins based on their names.
+   * @param pluginNames - The names of one or more plugins to deregister.
+   */
+  deregister(...pluginNames: string[]): Promise<Context>
+
+  get VERSION(): string
+}
+
+export async function load(settings: {
+  writeKey: string
+}): Promise<[AnalyticsNodeJs, Context]> {
+  const userOptions: UserOptions = {
+    disable: true,
+  }
+
+  const queue = new EventQueue(new PriorityQueue(3, []))
+  const options = { user: userOptions, group: userOptions }
+  const analytics = new Analytics(settings, options, queue)
+
+  const nodeSettings = {
+    writeKey: settings.writeKey,
+    name: 'analytics-node-next',
```

Comment:
might want to keep the name the same to avoid app changes (I think we have some custom renderers in the app debugger that are tied to library names). we can inject the `next` keyword in the version instead to keep the library name the same.

Reply:
You mean keep it the same as what `AnalyticsBrowser` uses or what analytics-node uses? This is the name that's being used today.

Reply:
Similarly do we maybe want to include node in the messageID? analytics-node uses `node-`, maybe something like `node-next-`? 😄